plugins/ltacX_common/g_ltacX_common.mli

```diff
@@ -0,0 +1,21 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+val goal_selector : Goal_select.t Pcoq.Entry.t
+
+val toplevel_selector : Goal_select.t Pcoq.Entry.t
+
+val wit_ltac_selector : (Goal_select.t, unit, unit) Genarg.genarg_type
+
+val ltac_selector : Goal_select.t Pcoq.Entry.t
+
+val wit_ltac_use_default : (bool, unit, unit) Genarg.genarg_type
+
+val ltac_use_default : bool Pcoq.Entry.t
```

Comment:
Maybe not for this PR, but this module should really go somewhere else that is part of Coq itself. All genargs are actually vernacular ones.

Reply:
Do you mean as a non plugin?

Reply:
Yes.

Reply:
It should be in this PR then, since we won't need the common plugin we won't need overlays.

Reply:
Why? It's not particularly about ltac IMO, it's more a generic tactic monad profiler.
I'm not doing a separate plugin just for profile_ltac.

Reply:
It's a crappy API that shouldn't be relied upon. When we come up with a cleaner one we'd have to reimplement it anyways.

Reply:
~~~
val do_profile_gen :
  ('a -> Pp.t option) -> 'a ->
  ?count_call:bool -> 'b Proofview.tactic -> 'b Proofview.tactic
~~~
looks pretty normal?

Also what does that have to do with it being in a plugin or not?

Reply:
About the API: it relies on printing the tactic being computed. This is both extremely slow and extremely fragile. We should have a way to register stack frames in a more principled fashion, e.g. with UIDs based on some dynamically extensible type or something.

About the plugin: it's easier to deprecate a plugin as a whole rather than piecewise modification of the clients that depend on it. Also, we're sending a message in one case w.r.t. the potential external users.

Reply:
I'm suprised to see that you think dealing with a plugin is easier than not dealing with a plugin. At least for me it is much easier to live plugin free. Since Ltac became a plugin I have witnessed 0 advantages to it, and many downsides.

Also I'm unsure that the messages we are sending about the API have a lot to do with plugins vs non-plugins.